store/bigtable/bigtable.go

```diff
@@ -0,0 +1,470 @@
+package bigtable
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/bigtable"
+	"github.com/grafana/metrictank/mdata"
+	"github.com/grafana/metrictank/mdata/chunk"
+	"github.com/grafana/metrictank/stats"
+	"github.com/grafana/metrictank/tracing"
+	opentracing "github.com/opentracing/opentracing-go"
+	tags "github.com/opentracing/opentracing-go/ext"
+	"github.com/raintank/schema"
+	log "github.com/sirupsen/logrus"
+)
+
+const Month_sec uint32 = 60 * 60 * 24 * 28
+
+var (
+	errChunkTooSmall  = errors.New("impossibly small chunk in bigtable")
+	errStartBeforeEnd = errors.New("start must be before end.")
+	errCtxCanceled    = errors.New("context canceled")
+
+	// metric store.bigtable.get.exec is the duration of getting from bigtable store
+	btblGetExecDuration = stats.NewLatencyHistogram15s32("store.bigtable.get.exec")
+	// metric store.bigtable.get.wait is the duration of the get spent in the queue
+	btblGetWaitDuration = stats.NewLatencyHistogram12h32("store.bigtable.get.wait")
+	// metric store.bigtable.put.exec is the duration of putting in bigtable store
+	btblPutExecDuration = stats.NewLatencyHistogram15s32("store.bigtable.put.exec")
+	// metric store.bigtable.put.wait is the duration of a put in the wait queue
+	btblPutWaitDuration = stats.NewLatencyHistogram12h32("store.bigtable.put.wait")
+	// metroc store.bigtable.get.error is the count of reads that failed
+	btblReadError = stats.NewCounter32("store.bigtable.get.error")
+
+	// metric store.bigtable.chunks_per_row is how many chunks are retrieved per row in get queries
+	btblChunksPerRow = stats.NewMeter32("store.bigtable.chunks_per_row", false)
+	// metric store.bigtable.rows_per_response is how many rows come per get response
+	btblRowsPerResponse = stats.NewMeter32("store.bigtable.rows_per_response", false)
+
+	// metric store.bigtable.chunk_operations.save_ok is counter of successful saves
+	chunkSaveOk = stats.NewCounter32("store.bigtable.chunk_operations.save_ok")
+	// metric store.bigtable.chunk_operations.save_fail is counter of failed saves
+	chunkSaveFail = stats.NewCounter32("store.bigtable.chunk_operations.save_fail")
+	// metric store.bigtable.chunk_size.at_save is the sizes of chunks seen when saving them
+	chunkSizeAtSave = stats.NewMeter32("store.bigtable.chunk_size.at_save", true)
+	// metric store.bigtable.chunk_size.at_load is the sizes of chunks seen when loading them
+	chunkSizeAtLoad = stats.NewMeter32("store.bigtable.chunk_size.at_load", true)
+	// metric store.bigtable.byte_size.at_save is the number of chunk bytes saved in each bulkApply
+	chunkBytesAtSave = stats.NewMeter32("store.bigtable.chunk_bytes.at_save", true)
```

Comment:
shouldn't this go with the other metrics under `store.bigtable.put` ?
e.g. `store.bigtable.put.bulk_bytes_saved` or something

Reply:
ditto for at_load -> get

Reply:
not for chunk_size.{at_load,at_save} because cass has those too.